pkg/app/piped/analysisprovider/metrics/prometheus/prometheus.go

```diff
@@ -83,20 +83,28 @@ func (p *Provider) Type() string {
 	return ProviderType
 }
 
-func (p *Provider) RunQuery(ctx context.Context, query string, queryRange metrics.QueryRange, evaluator metrics.Evaluator) (bool, string, error) {
+// Evaluate queries the range query endpoint and checks if values in all data points are within the expected range.
+// For the range query endpoint, see: https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries
+func (p *Provider) Evaluate(ctx context.Context, query string, queryRange metrics.QueryRange, evaluator metrics.Evaluator) (bool, string, error) {
 	if err := queryRange.Validate(); err != nil {
 		return false, "", err
 	}
 
 	ctx, cancel := context.WithTimeout(ctx, p.timeout)
 	defer cancel()
 
+	// NOTE: Use 1m as a step but make sure the "step" isn't smaller than the query range.
+	step := time.Minute
+	if diff := queryRange.To.Sub(queryRange.From); diff < step {
+		step = diff
+	}
+
 	p.logger.Info("run query", zap.String("query", query))
 	// TODO: Use HTTP Basic Authentication with the username and password when needed.
 	response, warnings, err := p.api.QueryRange(ctx, query, v1.Range{
 		Start: queryRange.From,
 		End:   queryRange.To,
-		Step:  queryRange.Step,
```

Comment:
So `queryRange.Step` will be ignored?

Reply:
Yeah, exactly. I settled on removing the `Step` field from `QueryRange` because it's used for only Prometheus currently: https://github.com/pipe-cd/pipe/pull/1749/files#diff-bc758c993e06bce2a8b71a6bd7749fbe3dcdfa77fe51fafc8b4f2c654cf8645aL51